packages/core/src/server.ts

```diff
@@ -582,6 +598,23 @@ export class Controller extends API {
         // authentication.
         if (!(await getCryptoProvider().timingSafeEqual(M, srp.M1!))) {
             this.log("account.createSession", { success: false });
+            ++srpState.failedAttempts;
+            await this.storage.save(auth);
+            if (srpState.failedAttempts >= 5) {
+                if (this.context.device) {
+                    try {
+                        await this.removeTrustedDevice(this.context.device.id);
+                    } catch (e) {}
+                }
+
+                if (acc.settings.notifications.failedLoginAttempts) {
+                    try {
+                        const location = this._buildLocationAndDeviceString(this.context.location, this.context.device);
+
+                        this.messenger.send(acc.email, new FailedLoginAttemptMessage({ location }));
+                    } catch (e) {}
+                }
+            }
```

Comment:
While testing this I noticed that you can keep trying indefinitely after 5 failed attempts and it will sent a notification for every single attempt. Instead, we should cancel the SRP session (by deleting it) so they have to at least go through the the email verification again before they can try again. You might need to add some code on the client to handle this though.

Reply:
I assume we'll delete the SRP session _even_ if we don't send the email, right?

Reply:
Yes, correct.